src/plugins/layout/nodeinfo.ts

```diff
@@ -20,5 +21,11 @@ function shouldUnmaskNode(node: Node, parentInfo: INodeInfo): boolean {
       ? (node as Element).getAttribute(UnmaskAttribute) === "true"
       : false
   );
-  return hasUnmaskAttribute || parentUnmasked;
+  // mask attribute takes precedence over any unmask above
+  const hasMaskAttribute = (
```

Comment:
I think you should change the logic here to inherit both the mask and unmask flags from the parent so that if you have a "mask" attribute (explicit) at a parent and there is no intervening unmask (and vice versa). Let's chat more if the comment isn't clear.

Reply:
making the change more clear

---
In reply to: [309714869](https://github.com/microsoft/clarity-js/pull/147#discussion_r309714869) [](ancestors = 309714869)